src/Constants_Conversion.f90

```diff
@@ -232,4 +236,130 @@ ELEMENTAL FUNCTION K_to_F(in_temp) RESULT(out_temp)
 
 ENDFUNCTION K_to_F
 !
+!-------------------------------------------------------------------------------
+!> @brief converts temperature to Kelvin from Celsius, Rankine, or Fahrenheit,
+!> if any other unit is presented in_temp is simply returned
+!> @param in_temp the temperature input as the specified units
+!> @param units character indicating the units of in_temp
+!> @return out_temp the temperature converted to Kelvin
+!>
+ELEMENTAL FUNCTION tempTo_K(in_temp,units) RESULT(out_temp)
+  REAL(SRK),INTENT(IN) :: in_temp
+  CHARACTER(LEN=1),INTENT(IN) :: units
+  REAL(SRK) :: out_temp
+
+  SELECTCASE(units)
+  CASE('C')
+    out_temp = C2K + in_temp
+  CASE('R')
+    out_temp = in_temp * R2K
+  CASE('F')
+    out_temp = (in_temp - 32.0_SRK)* R2K + 273.15_SRK
+  CASE DEFAULT
+    out_temp = in_temp
+  ENDSELECT
```

Comment:
So you should just call `K_to_C`, `K_to_F` and so on in all of these routines instead of duplicating the math here

Reply:
are we planning to replace the `K_to_C` etc interfaces with these, though?  If so, it's just going to be temporarily duplicated.

Reply:
I wasn't... If you know exactly what temperature something's in already, just use the direct conversion.  This is more for annoyances like input processing where there are multiple possibilities.  That's my opinion at least.

Reply:
Do we really want to deal with string matching for these interfaces?

Reply:
I second Ben's comment on string matching. Is enumeration possible?

Reply:
It's certainly possible... we just do it in exactly 0 places in Futility or MPACT.  We need the string version at the very least, but could add enumerations (`TEMP_K`, `TEMP_C`, etc.) and provide second versions of the interfaces as well.

Reply:
Ah so I was expecting we wouldn't keep the K_to_C, etc. If we do plan on it its an easy switch. As for using enumerations I can't think of a single reason not to, but I also don't really see a lot of benefit either. If all cases remain a single character and the only valid inputs are capital, then the select case is effectively identical.

Reply:
I"m okay having this interface IF we keep the old ones.  Most of the time you know what the units are and you can directly use `K_to_C` (or whatever).  I just didn't want a giant loop that is doing string matching a billion times.  Another option (definitely more complicated) is to have an interface that returns the function pointer based on the desired input and output units.  Honestly, if we keep `X_to_Y` I think this is fine.

Reply:
Alright cleaned up the duplicate math